tracks/python/exercises/say/mentoring.md

```diff
@@ -0,0 +1,127 @@
+    return result
+
+def say(number):
+    if 0 > number or number > 999999999999:
+        raise ValueError("input out of range")
+    result = []
+    for k, denomination in denominations:
+        if number > 10**k - 1:
+            result += say3(number // 10**k)
+            result += [denomination]
+            number %= 10**k
+    return " ".join(e for e in result if e) or "zero"
+```
+
+## Common Suggestions
+
+### Defining the literals
+Typically this problem will begin with defining the literals needed. 
+These can be done as either lists or dicts, but most students will fine dicts easier to work with for this problem. 
+Encourage the student to define these at the start rather than embedding them in their code.
+
+### Values below 1000
+Next, the instructions suggest begining by creating a function which works for values of up to three digits.
+The return value of this function will typically be a string, but it can also be a list of strings for later assembly.
+Whether in this function or for the final return, encourage them to use a list and combine the elements using `str.join`.
+
+The main challenges are ensuring that the 'hundred' denominator is conditionally included and that the tens and ones columns are joined with a hyphen if they are both present.
+Simple `if` checks for each of these conditions are the most straightforward approach.
+
+Another valid approach is to only handle values below 100, and instead handle the hundreds alongside the other denominators.
+This approach typically requires the use of recursion. 
+
+### Larger numbers
+Encourage the student to use the first function they wrote for each set of three digits rather than duplicating code.
+Each set of three can be checked individually and conditionally added along with the appropriate denominator, or alternatively the denominators can be stored in a list and iterated over.
+
+### Zero or invalid input
+Finally, the student must add a guard that returns a `ValueError` if the input is out of range. 
+A guard is also the easiest way to handle the zero case, but this can also be handled through short circuit evaluation.
+
+### Parsing the number
+The number needs to be broken up into groups of three. 
+This can be handled either with numeric operators or by casting to a string.
+Generally a student will find it easier to work with the number as a string for this problem, but both approaches are valid.
+
+## Talking points
+The student should be familiar with comprehensions - if they aren't, introduce the concept as they are very useful here.
+
+There are a wide variety of valid approaches to this problem, which gives students a lot of opportunity to experiment and encounter new ways of manipulating data.
+Some useful techniques that students might not have encountered are listed below.
+Students will not need or want to use all of these, but mentors should look for opportunities in the student's code to introduce one or more of them.
```

Comment:
I'm not sure this list of functions is super helpful when mentoring an exercise. As a mentor, I'd want to see discussions on approaches and solutions at a higher level, not specific string methods which may or may not be useful here.

Reply:
I do see your point about the specificity of some lines - 97, 99-101 are probably unnecessary. But since there's a lot going on in this exercise, I figured having a list of suggested techniques would be helpful as a quick reference - if a mentor scans this list before reading the student's code they'll more likely spot that a certain piece of code might make use of something here and be a good opportunity to introduce it to the student. Otherwise many of these wouldn't come to mind without more time spent on the exercise. I think it can be quite helpful to be introduced to a technique through a real example where it improves the code you've already written - getting a sense of when some of these would be useful is harder in contrived examples imo, but for slightly less common constructions it's easy to miss good opportunities to teach them.